src/worker/ingestion/process-event.ts

```diff
@@ -230,8 +231,17 @@ export class EventsProcessor {
                 `Could not find person with distinct id "${distinctId}" in team "${teamId}", even after trying to insert them`
             )
         }
+
         const updatedProperties: Properties = { ...propertiesOnce, ...personFound.properties, ...properties }
 
+        for (const [key, val] of Object.entries(incrementProperties)) {
+            if (updatedProperties[key]) {
+                updatedProperties[key] += val
+            } else {
+                updatedProperties[key] = val
+            }
+        }
```

Comment:
This may result in some race conditions, would be better to do in SQL if possible.

Reply:
hmm yes that's true - let me look into another approach

Reply:
Yep, this should normally be done inside SQL if possible. Googling gave something [like this](https://stackoverflow.com/questions/25957937/how-to-increment-value-in-postgres-update-statement-on-json-key/39076637):

```sql
UPDATE foo 
SET data = jsonb_set(data, '{bar}', (COALESCE(data->>'bar','0')::int + 1)::text::jsonb)
WHERE id = 1;
```

Reply:
Yeah I was looking into this as well - `jsonb` is great, but in CH we store properties as `varchar`...

Does give me an excuse to look at CH a bit more though

Reply:
Yeah, there will be no property updates in clickhouse... only a new insert with a new full array... :/

Reply:
Regarding clickhouse, since postgres is the master data store for persons, we can just do the update in postgres (inline like the query above), fetch the returned row (and its properties) and then store that in clickhouse. Theoretically nothing should override anything this way.